### PR TITLE
Add tanh Chebyshev approx

### PR DESCRIPTION
**Pull request by gmuraru**
This implementation is taken from the FacebookResearch Crypten project[1].
The PR that adds this functionality in Crypten is here[2].

[1] https://github.com/facebookresearch/CrypTen
[2] https://github.com/facebookresearch/CrypTen/pull/45

### CODE REVIEW

File: syft/frameworks/torch/tensors/interpreters/precision.py [hunk 1]
```diff
@@ -636,14 +639,30 @@ def log(tensor):
...
+            """
+            coeffs = syft.common.util.chebyshev_series(torch.tanh, maxval, terms)[1::2]
+            tanh_polys = syft.common.util.chebyshev_polynomials(tensor.div(maxval), terms)
+            tanh_polys_flipped = tanh_polys.unsqueeze(dim=-1).transpose(0, -1).squeeze(dim=0)
+
+            out = tanh_polys_flipped.matmul(coeffs)
+            # truncate outside [-maxval, maxval]
+            out = torch.where(tensor > maxval, 1.0, out)
+            out = torch.where(tensor < -maxval, -1.0, out)
+            return out
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/interpreters/precision.py]: I'm curious, `2 * sigmoid(2 * tensor) - 1`  wasn't precise enough or too slow?

Comment (gmuraru) [reply syft/frameworks/torch/tensors/interpreters/precision.py]: In the PR, they say this:
```
Implemented improved tanh approximation that's ~43.5% more accurate (measured by total relative error) and ~33% faster (see n196073).
```
Before this, they also had the implementation using the ```sigmoid```

Comment (LaRiffle) [reply syft/frameworks/torch/tensors/interpreters/precision.py]: Ok cool!

**Comment by gmuraru** [comment]
@iamtrask added the tests from (here)[https://github.com/facebookresearch/CrypTen/pull/45/files#diff-2706d8da110be452b14b8d05fa05a71e]
